code/backend/pages/01_Ingest_Data.py

```diff
@@ -122,6 +123,7 @@ def upload_file(bytes_data: bytes, file_name: str, content_type: Optional[str] =
             overwrite=True,
             content_settings=ContentSettings(content_type=content_type + charset),
         )
+        blob_client.set_blob_metadata({"source_url": file_name})
```

Comment:
added this for url embedding using pull not directly related to adding the env variable

Reply:
I'm not sure I understand what thius change is for.

Also, I think you can add the metadata in the `blob_client.upload_blob()` function call.